controllers/dspipeline_controller.go

```diff
@@ -266,17 +272,29 @@ func (r *DSPAReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 
 		err = r.ReconcileAPIServer(ctx, dspa, params)
 		if err != nil {
+			r.setStatusAsNotReady(config.APIServerReady, err, dspaStatus.SetApiServerStatus)
 			return ctrl.Result{}, err
+		} else {
+			r.setStatus(ctx, params.APIServerDefaultResourceName, config.APIServerReady, dspa,
+				dspaStatus.SetApiServerStatus, log)
```

Comment:
Pretty sure this is changing behavior of the original logic, can you take a look again at when `handleReadyCondition` is called, it is independent of whether Reconcilliation fails or succeed, it is a runtime check initiated when things like child deployments/pods experience change events

Reply:
This applies to PA and SWF as well

Reply:
In the original logic, it was called within `GenerateStatus`, which is called [at the end](https://github.com/opendatahub-io/data-science-pipelines-operator/blob/4dd4ef440609f3e67f30a78ec30100feb625286f/controllers/dspipeline_controller.go#L306) of the `Reconcile` function, just before updating the DSPA's status. Since the execution doesn't get to this point in case of errors, it was called only if no errors occurred.

If `handleReadyCondition` was originally intended for running on every reconciliation, it seems like, before my changes, the code no longer reflected this behavior.
I can fix that to run always. Can you just confirm it?

Reply:
you are absolutely right!